docs/source/app_developers_guide/zmq_event_subscription.rst

```diff
@@ -0,0 +1,309 @@
+********************************
+Using ZMQ to Subscribe to Events
+********************************
+
+Client applications can subscribe to Hyperledger Sawtooth events using
+`ZMQ <http://zeromq.org>`_ and protobuf messages. The general subscription
+process is as follows:
+
+#. Construct a subscription that includes the event type and an optional
+   filter or filters.
+
+#. Submit the event subscription as a message to the validator.
+
+#. Wait for a response from the validator.
+
+#. Start listening for events.
+
+This section summarizes event subscriptions, then explains the procedure for
+event subscription. It also describes the following operations:
+
+* Correlating events to blocks
+* Requesting event catch-up
+* Unsubscribing to events
+
+.. note::
+
+   This procedure uses Python examples to show how to subscribe to events. The
+   process is similar for any imperative programming language that meets these
+   requirements. A client application can use any language that provides a ZMQ
+   library and a protobuf library. In addition, the required Sawtooth protobuf
+   messages must be compiled for the chosen language.
+
+The following steps assume that the XO transaction family has a ``create`` event
+that is sent when a game has been created, as in this example:
+
+.. code-block:: python
+
+     context.add_event(
+         "xo/create", {
+             'name': name,
+             'creator', signer_public_key
+     })
+
+
+Step 1: Construct a Subscription
+================================
+
+An application can use the ``EventSubscription`` protobuf message to construct
+an event subscription. For example, in the ``sawtooth`` namespace, the
+application could subscribe to either a ``block-commit`` or ``state-delta``
+event (or both) in the ``sawtooth`` namespace, using either a ``SIMPLE`` or
+``REGEX`` filter.
+
+The following example constructs an event subscription for state-delta events
+(changes in state) with a ``REGEX_ANY`` filter for events from the XO
+transaction family.
+
+.. code-block:: python
+
+    subscription = EventSubscription(
+        event_type="sawtooth/state-delta",
+        filters=[
+            # Filter to only addresses in the "xo" namespace using a regex
+            EventFilter(
+                key="address",
+                match_string="xo.*",
```

Comment:
This is not the xo namespace. The xo namespace is `hashlib.sha512('xo'.encode("utf-8")).hexdigest()[0:6]` which comes out as `5b7349`, so the value here should be `match_string="5b7349.*"`